File: 2022/Day4/day4.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = __importDefault(require("fs"));
const path = require('path');
const filePath = path.join(__dirname, 'data.txt');
const file = fs_1.default.readFileSync(filePath, 'utf-8');
const inputArray = file.split('\n');
//iterate over inputArray
let item;
let first;
let second;
let range1Start;
let range1End;
let range2Start;
let range2End;
let sum = 0;
for (let i = 0; i < inputArray.length; i++) {
    item = inputArray[i].split(',');
    const [one, two] = item;
    first = one.split('-');
    second = two.split('-');
    range1Start = Number(first[0]);
    range1End = Number(first[1]);
    range2Start = Number(second[0]);
    range2End = Number(second[1]);
    // if ((range1Start >= range2Start && range1End <= range2End) || (range2Start >= range1Start && range1End >= range2End)){
    //     sum +=1;
    // }
    if ((range1Start <= range2Start && range1End >= range2Start) || (range2Start <= range1Start) && (range2End >= range1Start)) {
        sum += 1;
    }
}
console.log(sum);

File: 2022/Day4/day4.go
package main

import (
	"bufio"
	"log"
	"os"
	"strconv"
	"strings"
)

func main(){
	file, err := os.Open("/Users/johnwroge/advent_of_code/Day4/data.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()
	lines := []string{}
	scanner := bufio.NewScanner(file)

	for scanner.Scan(){
		line := scanner.Text()
		lines = append(lines, line)
	}
	sum := 0; 
	for i := 0; i < len(lines); i++ {
		item := strings.Split(lines[i], ",")
		item1 := item[0]
		item2 := item[1] 
		first := strings.Split(item1, "-")
		second := strings.Split(item2, "-")
	
		range1Start, err := strconv.Atoi(first[0]);
		if err != nil {
			log.Fatal(err)
		}
		range1End, err := strconv.Atoi(first[1]);
		if err != nil {
			log.Fatal(err)
		}
		range2Start, err := strconv.Atoi(second[0]);
		if err != nil {
			log.Fatal(err)
		}
		range2End, err := strconv.Atoi(second[1]);
		if err != nil {
			log.Fatal(err)
		}
		
			// if ((range1Start >= range2Start && range1End <= range2End) || (range2Start >= range1Start && range1End >= range2End)){
			// 	sum +=1;
			// }
			if ((range1Start <= range2Start) && (range1End >= range2Start)) || ((range2Start <= range1Start) && (range2End >= range1Start)){
            sum += 1
			}
	}
	print((sum))
}